### PR TITLE
Restrict inline properties

### PR DESCRIPTION
**Pull request by wwjhu**
Restricts the inline properties in the component's properties array to just inline header property definitions and references to existing component properties overriding the directive key.
```
{ control: { type: 'header' }, label: 'Header Label' },
```
```
{ "name": "hyperlink", "directiveKey": "hyperlink" }
```

Adds validation that verifies that parsed header properties have no dataType property

### CODE REVIEW

File: lib/components-schema-v1_7_x.ts [hunk 1]
```diff
@@ -35,6 +35,48 @@ function labelProperty(description: string): { oneOf: JSONSchema7Definition[] }
...
+            type: 'string',
+        },
+        {
+            type: 'object',
+            additionalProperties: false,
+            required: ['name', 'directiveKey'],
+            properties: {
+                name: {
+                    type: 'string',
+                    description: 'component property identifier',
```
Comment (jaapvanblaaderen) [inline lib/components-schema-v1_7_x.ts:50]: Small note: this description starts without capital, the other start with one. In the rest of the most descriptions start with a capital, though not all of them.

File: lib/components-schema-v1_7_x.ts [hunk 2]
```diff
@@ -399,45 +441,30 @@ const componentPropertyDefinition: {
...
-                        {
-                            type: 'object',
-                            additionalProperties: false,
-                            // Use a copy to avoid a circular reference and a stack overflow on childProperties at run-time
-                            properties: { ...componentPropertyDefinition },
-                        },
-                    ],
+                matchType: {
+                    type: 'string',
+                    description: `Defines how to match the parent property's value`,
```
Comment (jaapvanblaaderen) [inline lib/components-schema-v1_7_x.ts:452]: Property's reads a bit weird. Maybe: "Defines how to match the value of the parent property"?

File: lib/components-schema-v1_7_x.ts [hunk 3]
```diff
@@ -399,45 +441,30 @@ const componentPropertyDefinition: {
...
-            matchType: {
-                type: 'string',
-                description: `Defines how to match the parent property's value`,
-                enum: ['exact-value'],
-            },
-            matchExpression: {
-                type: ['boolean', 'integer', 'string', 'number'],
-                description: `The expression to use to match the parent property's value`,
-            },
+    childProperties: {
```
Comment (jaapvanblaaderen) [inline lib/components-schema-v1_7_x.ts:444]: Not really sure how this is done in the rest of the validator, but are/should there be tests to validate the schema? For example testing with a definition that contains an invalid `matchType`.

**Review by jaapvanblaaderen — APPROVED** [review]
LGTM, just a few remarks / questions

File: lib/components-schema-v1_7_x.ts (hunk 3 above)
Comment (wwsno) [reply lib/components-schema-v1_7_x.ts:444]: I think we have tests in https://github.com/WoodWing/studio-component-set-tools/pull/93.

Comment (wwjhu) [reply lib/components-schema-v1_7_x.ts:444]: Positive flow test is indeed in #93, this PR restricts the possibilities. The question is what negative tests would bring us. Wouldn't that be just testing whether `AJV` does its job?

Comment (jaapvanblaaderen) [reply lib/components-schema-v1_7_x.ts:444]: Good point.

File: lib/components-schema-v1_7_x.ts [hunk 4]
```diff
@@ -35,6 +35,48 @@ function labelProperty(description: string): { oneOf: JSONSchema7Definition[] }
...
 
+const inlineComponentPropertyDefinitionOrReferenceList: { oneOf: JSONSchema7Definition[] } = {
+    oneOf: [
+        {
+            type: 'string',
+        },
+        {
+            type: 'object',
+            additionalProperties: false,
+            required: ['name', 'directiveKey'],
```
Comment (wwsno) [inline lib/components-schema-v1_7_x.ts:46]: https://github.com/WoodWing/csde-components-boilerplate/blob/master/components/components-definition.json#L34

So we can also allow:
`{ "name": "prop", "defaultValue": "value" }`

Comment (wwjhu) [reply lib/components-schema-v1_7_x.ts]: Added.

**Review by wwsno — APPROVED** [review]

**Pull request merged**